tests/test_byteutils.nim

```diff
@@ -65,6 +67,17 @@ suite "Byte utils":
     expect(ValueError): discard hexToByteArray[1]("")
     expect(ValueError): discard hexToByteArray[1]("1")
 
+  test "hexToByteArray: Invalid value":
+    expect(ValueError):
+      var buffer: array[1, byte]
+      hexToByteArray("0x", buffer)
+
+    expect(ValueError):
+      var buffer: distinct seq[byte]
+      hexToByteArray("0x", (openArray[byte])buffer)
```

Comment:
This line doesn't seem well-formed. It seems a bug to me that you didn't get a compiler error.

If you wan to convert a distinct `seq` type to an open array, the idiomatic code would be to use the `distinctBase` operator and to let the compiler perform an implicit conversion to `openArray`.

Reply:
Ok sure, this is from C++ style days, I've seen it around a few nim repos

```nim
openArray[byte](buffer)
(openArray[byte]) buffer
```
Are all valid. I will switch to `distinctBase`

Reply:
Talked with @zah in discord and cleared the use of this as I would like it to be very clear what the problem is here.